website/static/js/contribRemover.js

```diff
@@ -0,0 +1,294 @@
+/**
+ * Controller for the Remove Contributor modal.
+ */
+'use strict';
+
+var $ = require('jquery');
+var ko = require('knockout');
+var bootbox = require('bootbox');
+var Raven = require('raven-js');
+
+var oop = require('./oop');
+var $osf = require('./osfHelpers');
+var Paginator = require('./paginator');
+
+var API_BASE = window.contextVars.apiV2Prefix;
+
+function getNodesOriginal(nodeTree, nodesOriginal) {
+    /**
+     * take treebeard tree structure of nodes and get a dictionary of parent node and all its
+     * children
+     */
+    var i;
+    var j;
+    var adminContributors = [];
+    var registeredContributors = [];
+    var nodeId = nodeTree.node.id;
+    for (i=0; i < nodeTree.node.contributors.length; i++) {
+        if (nodeTree.node.contributors[i].is_admin) {
+            adminContributors.push(nodeTree.node.contributors[i].id);
+        }
+        if (nodeTree.node.contributors[i].is_confirmed) {
+            registeredContributors.push(nodeTree.node.contributors[i].id);
+        }
+    }
+    nodesOriginal[nodeId] = {
+        public: nodeTree.node.is_public,
+        id: nodeTree.node.id,
+        title: nodeTree.node.title,
+        contributors: nodeTree.node.contributors,
+        isAdmin: nodeTree.node.is_admin,
+        visibleContributors: nodeTree.node.visible_contributors,
+        adminContributors: adminContributors,
+        registeredContributors: registeredContributors
+    };
+
+    if (nodeTree.children) {
+        for (j in nodeTree.children) {
+            nodesOriginal = getNodesOriginal(nodeTree.children[j], nodesOriginal);
+        }
+    }
+    return nodesOriginal;
+}
+
+var RemoveContributorViewModel = oop.extend(Paginator, {
```

Comment:
`RemoveContributorViewModel` needn't extend `Paginator`. It can be a plain old JavaScript function.